bundles/org.openhab.binding.nanoleaf/src/main/java/org/openhab/binding/nanoleaf/internal/handler/NanoleafControllerHandler.java

```diff
@@ -514,18 +681,29 @@ private void sendStateCommand(String channel, Command command) throws NanoleafEx
                 } else if (command instanceof IncreaseDecreaseType) {
                     // increase/decrease brightness
                     if (controllerInfo != null) {
-                        Brightness brightness = controllerInfo.getState().getBrightness();
-                        if (command.equals(IncreaseDecreaseType.INCREASE)) {
-                            brightness.setValue(Math.min(brightness.getMax(),
-                                    brightness.getValue() + BRIGHTNESS_STEP_SIZE));
+                        @Nullable Brightness brightness = controllerInfo.getState().getBrightness();
+                        int brightnessMin = 0;
+                        int brightnessMax = 0;
+                        if (brightness!=null) {
+                            @Nullable Integer min = brightness.getMin();
+                            brightnessMin = (min == null) ? 0 : min;
+                            @Nullable Integer max = brightness.getMax();
+                            brightnessMax = (max == null) ? 0 : max;
+
+                            if (command.equals(IncreaseDecreaseType.INCREASE)) {

```

Comment:
Position literals first in `String` comparisons - that way if the `String` is `null` you won't get a `NullPointerException`, it'll just return `false`. [source: PMD](http://pmd.sourceforge.net/pmd-4.3.0/rules/design.html#PositionLiteralsFirstInComparisons)

Reply:
Apply where needed